doc/chassis/fabric.md

```diff
@@ -0,0 +1,163 @@
+    “fec_errors”: {{number}}
+    “fec_errors_corrected”: {{number}}
+```
+
+Note that the FABRIC_PORT_TABLE will also have entries in the Linecard Sonic instances because there are fabric ports in each forwarding ASIC as well.
+
+## 2.3 System Initialization
+
+As part of multi-ASIC support, /etc/sonic/generated_services.conf contains the list of services which will be created for each asic when the system boots up. This is read by systemd-sonic-generator to generate the service files for each container that needs to run. 
+
+Since the fabric ASIC doesn’t need lldp, bgpd and teamd containers to run, systemd-sonic-generator will be modified to not start these services for the fabric ASICs.
+
+## 2.4 Fabric Card Hotswap
+
+PMON will be responsible for detecting card presence and hotswap events using the get_change_event API. A new systemd service will be responsbile for turning on/off the service files for the syncd, database and swss containers that manage each fabric ASIC. When the fabric card is removed, the containers that manage the fabric ASICs that are part of that fabric card will be stopped. These will be re-started when the fabric card is inserted later.
+
+## 2.5 Orchagent
+
+Orchagent creates the switch using the SAI API similar to creating the switch for a forwarding ASIC, except that the switch type will be fabric. When the ASIC is initialized, all the fabric ports are initialized by default. The fabric ports are a subtype of SAI Port object and it can be obtained by getting all the fabric port objects from SAI. Since there are no front panel ports on a fabric ASIC, port_config.ini will be empty and portsyncd will not run. 
+
+On fabric ASICs, OrchDaemon will only monitor and manage fabric ports. It will not maintain cpu port and front panel port related ochres, such as PortsOrch, IntfsOrch, NeighborOrch, VnetOrch, QosOrch, TunnelOrch, and etc. To simplify the change, we will just create FabricOrchDaemon inheriting OrchDaemon for fabric ASICs and this will only run FabricPortsOrch, the module responsible for managing fabric ports.
+
+## 2.6 Fabric Ports in Forwarding ASICs
+
+When a forwarding ASIC is initialized, the fabric ports are initialized by default by SAI. Orchagent will run FabricPortsOrch in addition to all the other orchs that needs to be run to manage the forwarding ASIC. Fabric port monitoring and handling is identical to what happens on a Fabric ASIC.
+
+## 2.7 Cli commands
+
+```
+> show fabric counters [asic_name] [port_id]
+
+asic2 fabric port counter (number of fabric ports: 192)
+
+PORT            RxCells     TxCells      Crc       Fec  Corrected
+-------------------------------------------------------------------------
+ 0           : 71660578         2         0         0         0
+ 1           : 71659798         1         0         0       213
+ 2           :        0         1         0         0       167
+ 3           :        0         2         0         0       193
+```
+
+### 2.7.1 Fabric Status
```

Comment:
As there can be many fabric links in the chassis, may be even more useful if there is a CLI to only display the links/fabric ports that have errors.  Eg.. "show fabric counters errors"